android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/FacebookFeedFragmentViewModel.java

```diff
@@ -36,6 +36,10 @@ public FacebookFeedFragmentViewModel() {
         if (fbPageId == null) {
             Publisher<FacebookPageId> publisher = FacebookApi.getFacebookGraphAPI().getPageId(SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null),
                     fbAccessToken)
+                    .onErrorReturn(throwable -> {
+                        Timber.e(throwable);
+                        return new FacebookPageId(null, FacebookPageId.INVALID);
+                    })
```

Comment:
It's not necessary that each error is about facebook page ID. Detect the specific error and throw only if matched

Reply:
@iamareebjamal If the error is not related to 404 then what should we return? because UI should be informed that error has occurred and can stop swipe refresh layout.

Reply:
Let the error propagate as usual

Reply:
because we are wrapping publisher in live data we can not propagate error (that's the reason for crashing), we need to handle it there only. 
we can wrap the facebookPageId in Resource live data that can contain data, status, error message as used in Google's sample on Android Architecture components https://github.com/googlesamples/android-architecture-components/blob/master/GithubBrowserSample/app/src/main/java/com/android/example/github/vo/Resource.java.

Reply:
Use a wrapper